tests_profile

```diff
@@ -0,0 +1,69 @@
+# - JAVA_LIB_DIR - path to libjvm.
+# - RUST_LIB_DIR - path to Rust stdlib.
+# - RUST_COMPILER_VERSION - Rust version for building tests.
+# - RUSTFLAGS - Rust build options augmented with "rpath" option for JAVA_LIB_DIR and RUST_LIB_DIR.
+#   These options will make the linker add paths to the libjvm and the Rust standard library
+#   to the runtime search paths.
+#
+#   The runtime search paths, set by "rpath" option, work similarly on Linux and MacOS.
+#   On Linux the linker adds the entries to RUNPATH section in the binary, which is scanned
+#   after LD_LIBRARY_PATH. [1]
+#   On MacOS the linker adds load commands with paths to the libraries to the end
+#   of the runtime search list. Therefore, the load commands for these libraries
+#   *at the beginning* (@rpath/libNAME.dylib entries) will succeed in case the library is
+#   available in the default system directories or in LD_LIBRARY_PATH. [2]
+#
+#   If you need to override the libjvm, set LD_LIBRARY_PATH to specify the path to it.
+#
+#   "rpath" option in RUSTFLAGS variable affects all executables and dynamic libraries
+#   we build: the application, integration tests, libjava_bindings shared library.
+#
+#   It is worth mentioning that recent releases of MacOS with enabled by default SIP
+#   clear LD_LIBRARY_PATH when new shell started.
+#   Such behavior greatly affects Maven and all that it runs.
+#   Setting rpath is a good way to fix this flaw in tests.
+#   On Linux, it is not important whether to prefer RUNPATH or LD_LIBRARY_PATH.
+#
+# [1] https://amir.rachum.com/blog/2016/09/17/shared-libraries/#runtime-search-path
+# [2] https://github.com/aidansteele/osx-abi-macho-file-format-reference#header-structure-and-load-commands
+
+
+# Use the Java that Maven uses.
+#
+# Unfortunately, a simple `which java` will not work for some users (e.g., jenv),
+# hence this a bit complex thing.
+export JAVA_HOME="$(mvn --version | grep 'Java home' | sed 's/.*: //')"
+echo "JAVA_HOME=${JAVA_HOME}"
+
+# Find the directory containing libjvm (the relative path has changed in Java 9).
+export JAVA_LIB_DIR="$(find ${JAVA_HOME} -type f -name libjvm.\* | xargs -n1 dirname)"
+echo "JAVA_LIB_DIR=${JAVA_LIB_DIR}"
+
+# Version of Rust used to build tests.
+## Stable works well unless you want benchmarks.
+# TODO: stable does not work well until ECR-1839 is resolved
+export RUST_COMPILER_VERSION="${RUST_COMPILER_VERSION:-1.26.2}"
+echo "RUST_COMPILER_VERSION: ${RUST_COMPILER_VERSION}"
+
+# Find the directory containing Rust libstd.
+export RUST_LIB_DIR=$(rustup run ${RUST_COMPILER_VERSION} rustc --print sysroot)/lib
+echo "RUST_LIB_DIR: ${RUST_LIB_DIR}"
+
+# Linking options with additional paths for the application and tests.
+# This may override previous `-C link-args` if it appears in RUSTFLAGS before.
+RUSTFLAGS="${RUSTFLAGS:-}"
+if [[ $RUSTFLAGS = *"-C link-args="* ]]; then
+    echo "Error: the '-C link-args' RUSTFLAGS option will be overridden. Merge is not yet supported."
```

Comment:
We discovered a problem with this code on Friday: you will usually run this script in your terminal so that you can use `mvn foo` of `cargo baz`. But the second time you use it (directly or indirectly), it will close your terminal :smirk_cat:

Reply:
Let's just print warning message then

Reply:
Yep, that's probably the sanest option we have.